### DISCUSSION TITLE
Resolving Precedence for Style Shorthand Props

### DISCUSSION THREAD

**Discussion by MrAlexLau** [chakra-ui/chakra-ui#8497] [Q&A] [answered]
I noticed some interesting behavior while mixing style shorthand props with their long form counterparts (eg- `bg` vs `backgroundColor`).

If I'm referencing a Chakra component directly, the precedence makes sense to me. When both the shorthand and long form style props are present, whichever comes last takes precedence:
```javascript 
// Renders with a blue background
<Text backgroundColor="red" bg="blue">
  hi
</Text>

// Renders with a red background
<Text bg="blue" backgroundColor="red">
  hi
</Text>
```

However, the precedence becomes less clear when the value for these attributes is a responsive styling hash:

```javascript 
// Renders blue by default and green on a large viewport
<Text
  backgroundColor={{
    base: 'red',
    lg: 'green',
  }}
  bg="blue"
>
  hi
</Text>
```

While I think I understand what's happening here (`bg="blue"` does not overwrite the `backgroundColor -> lg` prop since the media query adds more specificity to the underlying css rule), I can see how others would find that to be unintuitive. Now I realize this case is contrived - you wouldn't normally be placing `backgroundColor` and `bg` props next to each other as I've done. 

However it's a very plausible scenario that an application would have a custom component that builds upon a Chakra component and spreads out the props given to it. In this case, the implications of the name collisions are more troublesome. For example, if I have a component like this:

```javascript
const MyCustomText = props => (
  <Text
    backgroundColor={{
      base: 'red',
      lg: 'green',
    }}
    {...props}
  />
);
```

Then any consumer of this component needs to be aware that `backgroundColor` is the "right" prop to override. Suppose my goal is to make the background always be blue:

```javascript
// This won't work for the same reason as before, on lg viewports `backgroundColor -> lg` has higher precedence
<MyCustomText bg="blue" /> 

// This will work as expected, since the prop name matches
<MyCustomText backgroundColor="blue" />
```

Admittedly I could add extra logic to `MyCustomText` to coerce the `bg` prop into `backgroundColor`, but that would be a lot of extra work to account for _any arbitrary_ given shorthand prop name.

Does this all make sense? Is it reasonable behavior to propose that the shorthand prop name coercion happens at the Chakra level?

**Comment by segunadebayo** [comment, marked as answer]
Later style props win. A longhand does not automatically beat its shorthand.

Put overrides last, or put them in `css` so they land after the rest.
